### PR TITLE
Add support for SetScanout

### PR DESCRIPTION
**Pull request by dorindabassey**
Add support for SetScanout

### CODE REVIEW

File: staging/vhost-device-gpu/src/virt_gpu.rs [hunk 1]
```diff
@@ -601,6 +614,26 @@ impl VirtioGpu {
...
+    fn update_scanout_resource(
+        &mut self,
+        _scanout_id: u32,
+        scanout_data: Option<VirtioScanoutBlobData>,
+        resource_id: u32,
+    ) -> VirtioGpuResult {
+        // Virtio spec: "The driver can use resource_id = 0 to disable a scanout."
+        if resource_id == 0 {
+            return Ok(OkNoData);
+        }
```
Comment (mtjhrc) [inline staging/vhost-device-gpu/src/virt_gpu.rs]: This wouldn't actually disable the scanout. Which is fine for now, so maybe add a coment like `// TODO: implement disabling scanout` or maybe even log an error that it is not implemented just so we know we ran into it.

File: staging/vhost-device-gpu/src/vhu_gpu.rs [hunk 2]
```diff
@@ -166,8 +167,9 @@ impl VhostUserGpuBackend {
                 virtio_gpu.resource_create_3d(resource_id, resource_create_3d)
             }
             GpuCommand::ResourceUnref(info) => virtio_gpu.unref_resource(info.resource_id),
-            GpuCommand::SetScanout(_info) => {
-                panic!("virtio_gpu: GpuCommand::SetScanout unimplemented");
+            GpuCommand::SetScanout(info) => {
+                debug!("SetScanout: {info:?}");
+                virtio_gpu.set_scanout(info.scanout_id, info.resource_id, None)
```
Comment (mtjhrc) [inline staging/vhost-device-gpu/src/vhu_gpu.rs]: Here `None` is passed as the `scanout_data` argument. But info contains things (namely a `virtio_gpu_rect` that has `width`/`height`/...) that should probably be stored with the resource.

**Review by mtjhrc — COMMENTED** [review]
Like I said, it seems we should call `VHOST_USER_GPU_SCANOUT` like the C vhost-user-gpu does..

I am quite unsure what parameters should we store about the resource  (`width`/`height` ...?), and in which struct.  But I think we may have to tweak that once we understand how other commands interact with this.

File: staging/vhost-device-gpu/src/vhu_gpu.rs (hunk 2 above)
Comment (dorindabassey) [reply staging/vhost-device-gpu/src/vhu_gpu.rs]: well for crosvm i see that they pass none for `scanout_data` in SetScanout gpucommand and pass something for set_scanout in SetScanoutBlob gpucommand, so which means that info should not contain any data in width or height which is ``` SetScanout: virtio_gpu_set_scanout { r: virtio_gpu_rect { x: 0, y: 0, width: 640, height: 480 }, scanout_id: 0, resource_id: 0 } ``` probably they expect that SetScanoutBlob to accept some values

Comment (mtjhrc) [reply staging/vhost-device-gpu/src/vhu_gpu.rs]: Hmm, interesting. I was looking at the C vhost-user-gpu and there they store the width and height. Not sure if we need to store it, I guess we will see if we need it later. But we should definitely send the `width`/`height` to QEMU using the `GpuBackend`.

**Review by mtjhrc — APPROVED** [review]
Looks good now!

**Pull request merged**